### DISCUSSION TITLE
How to write constrains of "or" with yalmip

### DISCUSSION THREAD

**Discussion by Leonyn1** [yalmip/YALMIP#1176] [Optimization modelling] [answered]
for example, I want to model the following equations:
`x1 - x2 >= 1 or x2 - x1 >= 1`
My current solution is that:
I create 0-1 variables x3, x4
`x1 - x2 - 1 + max_num * (1 - x3) >= 0`
`x2 - x1 - 1 + max_num * (1 - x4) >= 0`
`x3 + x4 >= 1`
among them, max_num is a very large number that bigger than `|x1| + |x2|`

I want to ask that is there better way to model this situation?
thanks for hearing my question.

**Comment by johanlofberg** [comment, marked as answer]
Yes, that looks like the vanilla big-M model for or (i.e. essentially what YALMIP implements when you use | (explitcily using function or))

**Reply by Leonyn1** [reply]
Thank you for answering my question!
My understanding is that I can directly use | to implement the model above, but how?
or how do you implements or with yalmip?
thanks again!

**Reply by johanlofberg** [reply]
or(x1-x2>= 1, x2-x1>=1) alternatively (x1-x2>= 1) | (x2-x1>=1) alternatively and most easily abs(x1-x2)>=1

In all cases explicit bound constraints are needed in the model on all variables for the big-M model to be deried
